### PR TITLE
Update Cache RedisHandler to support select database.

### PR DESCRIPTION
**Pull request by vibbow**
**Description**
Update Cache RedisHandler to support select database.
As this feature already exist in Session RedisHandler

**Checklist:**
- [ Y ] Securely signed commits
- [ N/A ] Component(s) with PHPdocs
- [ N/A ] Unit testing, with >80% coverage
- [ Y ] User guide updated
- [ Y ] Conforms to style guide

### CODE REVIEW

File: system/Cache/Handlers/RedisHandler.php [hunk 1]
```diff
@@ -118,6 +119,11 @@ public function initialize()
 			{
 				//              log_message('error', 'Cache: Redis authentication failed.');
 			}
+
+			if (isset($config['database']) && ! $this->redis->select($config['database']))
+			{
+				//              log_message('error', 'Cache: Redis select database failed.');
```
Comment (lonnieezell) [inline system/Cache/Handlers/RedisHandler.php]: This shouldn't be commented out.

Comment (lonnieezell) [reply system/Cache/Handlers/RedisHandler.php]: Perfect. Thanks! Merging.

Comment (vibbow) [reply system/Cache/Handlers/RedisHandler.php]: I had commented this out to log an error message.

But in the mean time, I notice only `redis->connect` will throw an RedisException is connection is failed.

`redis->auth` will only return a false rather an RedisException if auth is failed.

Should I enable the log for that either?

Comment (lonnieezell) [reply system/Cache/Handlers/RedisHandler.php]: Oh, yes, we should definitely be logging all of those situations. Though I've already merged this PR so you'll probably have to make another. Thanks!

**Pull request merged**